test/index.js

```diff
@@ -547,6 +547,30 @@ describe('Nightmare', function () {
       value.should.equal('');
     })
 
+    it('should not type in a nonexistent selector', function*(){
+      var failed = false;
+      try {
+        yield nightmare
+          .goto(fixture('manipulation'))
+          .type('does-not-exist', 'nightmare');
+      } catch(e){
+        failed = true;
+      }
+      failed.should.be.true;
```

Comment:
I wonder if using [Chai-as-promised](http://chaijs.com/plugins/chai-as-promised/) might be a good a way to clean these up. A bunch of the `goto` tests have similar reverse-assertion logic, and could be more easily written:

```js
it('should not type in a nonexistent selector', function() {
  return nightmare
    .goto(fixture(‘manipulation’))
    .type(‘does-not-exist’, ‘nightmare’)
    .then().should.be.rejected;
});
```